### ISSUE TITLE
When converting to flac, clicks or overloads appear, although converting to wav does not.

### ISSUE THREAD

**Issue by IlyaIndigo** [FluidSynth/fluidsynth#696]
### FluidSynth version
FluidSynth runtime version 2.1.5
Copyright (C) 2000-2020 Peter Hanappe and others.
Distributed under the LGPL license.
SoundFont(R) is a registered trademark of E-mu Systems, Inc.

FluidSynth executable version 2.1.5
Sample type=double

### Describe the bug
Midi file WarCraft TitleTheme.mid get  https://freemidi.org/getter-9265

fluidsynth -F WarCraft.wav /usr/share/sounds/sf2/FluidR3_GM.sf2 TitleTheme.mid
Converting from midi to wav is fine.

fluidsynth -F WarCraft.flac /usr/share/sounds/sf2/FluidR3_GM.sf2 TitleTheme.mid
But when converting to flac, clicks appear at loud sections, as if overload.
The behavior is very strange.

### Expected behavior
The sound of flac should be no different from the sound of wav.

### Steps to reproduce
Download midi file https://freemidi.org/getter-9265 (wget don't download, need browser)
Install fluidsynth and GM soundfont.
Convert to wav: fluidsynth -F WarCraft.wav /usr/share/sounds/sf2/FluidR3_GM.sf2 TitleTheme.mid
Convert to flac: fluidsynth -F WarCraft.flac /usr/share/sounds/sf2/FluidR3_GM.sf2 TitleTheme.mid
Play any player, mpv for example, first wav, then flac and hear clicks or overdrive on flac.

**Comment by derselbst** [comment]
Cannot reproduce. Please upload the broken flac file. Also, which version of libsndfile are you using?

**Comment by IlyaIndigo** [comment]
[WarCraft-broken.flac.zip](https://github.com/FluidSynth/fluidsynth/files/5432625/WarCraft-broken.flac.zip)
libsndfile1 1.0.28

**Comment by IlyaIndigo** [comment]
In mpd, clicks are not so noticeable. but they are still there.
But in mpv they are immediately noticeable.

**Comment by IlyaIndigo** [comment]
If it is confirmed that this is due to patches to libsndfile, open the openSUSE report.

**Comment by derselbst** [comment]
The MIDI file is too loud. See this printscreen of audacity. The flac file on top, the wave on bottom:

![Screenshot_20201024_112637](https://user-images.githubusercontent.com/8152480/97078327-e8132a00-15eb-11eb-92e5-6cbd9399d249.png)

In both cases, you see the clipping. The wave and flac files generated by fluidsynth both contain 16bit integer PCM. When imported into audacity, both files are converted to 32bit floating point. Most likely by using `sf_read_float`. During this process, the clipping behaviour apparently differs for both formats. This might have to do with the normalisation performed during the conversion, see http://www.mega-nerd.com/libsndfile/api.html#note1. But it doesn't explain why the behaviour differs between the two formats. You could ask Meganerd for details.

Some audio players seem to perform the same "to-float" conversion. My player doesn't, so I didn't notice the clippings first. Fun fact: I'm on openSUSE as well.

Long story short: adjust fluidsynth's gain to get rid of clipping by passing e.g. `-g 0.1`

**Comment by IlyaIndigo** [comment]
I never thought that a midi file could be too loud, because it's just a set of instructions for a synthesizer.
With -g0.1, of course it works, as well as converting without decay to wav and converting to flac.
Please tell me which way is still more correct, from the point of view of sound accuracy, to lower the gain and re-convert to flac, or to re-convert as it is in wav, and then re-convert to flac?

**Comment by derselbst** [comment]
> I never thought that a midi file could be too loud

Ok, let me correct: The MIDI + Soundfont + Synthesizer combination results in a too-high audio amplitude.

From the point of "sound accuracy", your primary goal should be to get rid of the clipping. Either by lowering the master gain or using a different soundfont or overriding the default modulators so that the NoteOn Velocity results in a more attenuated signal, etc.

The secondary goal should be to avoid any lossy conversions. While converting wav to flac is lossless, converting from 16bit int to 32bit float is not (due to the finite accuracy of IEEE754). Unfortunately, you rarely know about this latter conversion, as you have to rely on the programmers of audio players and editors. On the other hand, this latter conversion is negligible in 99,9% of cases... until we come across an issue like this one where it's not...

**Comment by IlyaIndigo** [comment]
Thank you for your reply.
